### ISSUE TITLE
[FEATURE] Apple Music Support

### ISSUE THREAD

**Issue by Red6785** [nathom/streamrip#697]
### Is the feature request related to a problem. Please describe it.

I know that this has been requested before, and denied due to a lack of open source decryption projects, but that has changed. Is it possible to take inspiration from [this project](https://github.com/WorldObservationLog/AppleMusicDecrypt) and hopefully get Apple Music support inside of StreamRip?

### Describe the solution you would like.

If possible, I'd like to see Apple Music downloading added to StreamRip using decryption inspired by this project.

### Describe alternatives you've considered.

An alternative would be to just use the project, however, I chose StreamRip because it's an All-In-One for most of the services I use.

**Comment by nathom** [comment]
Interesting. I'll put it on the roadmap.

**Comment by puellabyte** [comment]
> Interesting. I'll put it on the roadmap.

Worth mentioning if you have yet to look into it that this requires emulating an android device to perform the decryption.
While it would be nice to have as a feature to streamrip, the implementation would be hell. 

While you could use qemu and provide an .avd file yourself pre-configured with the required apps, signed into apple music on a premium subbed account, the unfortunate situation is that decryption will break if two users attempt to do it at the same time on the same account, meaning if you instead chose to run a server that performed the decryption for everybody using a single account a queue would have to be implemented.

I'm also going to assume you don't intend to create an entire GUI for the kvm to navigate signing into the apple music app.

**Comment by nathom** [comment]
> > Interesting. I'll put it on the roadmap.
> 
> 
> 
> Worth mentioning if you have yet to look into it that this requires emulating an android device to perform the decryption.


Yeah I saw. If the situation remains like that I won't be including it. I'll still leave the issue open in case someone figures out how to get tracks through the API.

**Comment by JFizDaWiz** [comment]
could always implement the feature but rely on the user providing their own login

**Comment by Red6785** [comment]
1. Repository I linked to has been abandoned. [This repository](https://github.com/zhaarey/apple-music-alac-atmos-downloader) is still in development, I believe.
2. This is how it works, @JFizDaWiz. The issue is not about accounts, the issue is that, as of current, the only way to rip the music from Apple Music is to set up Android Studio, with a specific version of Android and a specific version of Apple Music. It's very complex, compared to just exposing the API.

**Comment by yepitro** [comment]
hope this available soon
